server/resources/resource_management.go

```diff
@@ -1013,6 +1069,86 @@ func ActionResource(context middleware.Context, dataStore db.DB, identityService
 	return fmt.Errorf("no response")
 }
 
+func validateAttachment(
+	context middleware.Context,
+	policy *schema.Policy,
+	auth schema.Authorization,
+	resourceSchema *schema.Schema,
+	dataMap map[string]interface{},
+) error {
+	relationPropertyName := policy.GetRelationPropertyName()
+	if relationPropertyName == "*" {
+		for i := range resourceSchema.Properties {
+			prop := &resourceSchema.Properties[i]
+			if err := validateAttachmentRelation(context, policy, auth, dataMap, prop); err != nil {
+				return err
+			}
+		}
+		return nil
+	} else {
+		prop, err := resourceSchema.GetPropertyByID(relationPropertyName)
+		if err != nil {
+			return err
+		}
+		return validateAttachmentRelation(context, policy, auth, dataMap, prop)
+	}
+}
+
+func validateAttachmentRelation(
+	context middleware.Context,
+	policy *schema.Policy,
+	auth schema.Authorization,
+	dataMap map[string]interface{},
+	property *schema.Property,
+) error {
+	manager := schema.GetManager()
+	if property.Relation == "" {
+		// Not a relation, so skip
+		return nil
+	}
+
+	relatedSchema, _ := manager.Schema(property.Relation)
+	relatedResourceID, _ := dataMap[property.ID].(string)
+	if relatedResourceID == "" {
+		return nil
+	}
+
+	otherCond := policy.GetOtherResourceCondition()
+	tenants := otherCond.GetTenantIDFilter(schema.ActionRead, auth.TenantID())
+	filter := transaction.IDFilter(relatedResourceID)
+	if tenants != nil {
+		filter["tenant_id"] = tenants
+	}
+	otherCond.AddCustomFilters(filter, auth.TenantID())
+
+	options := &transaction.ViewOptions{}
+
+	mainTransaction := context["transaction"].(transaction.Transaction)
+	relatedRes, err := mainTransaction.Fetch(relatedSchema, filter, options)
+
+	errorNotFound := func() error {
+		return goext.NewErrorNotFound(errors.Errorf("Related resource of ID \"%s\" not found", relatedResourceID))
+	}
+
+	if err != nil {
+		if policy.IsDeny() {
+			return nil
+		} else {
+			return errorNotFound()
+		}
+	}
+
+	err = otherCond.ApplyPropertyConditionFilter(schema.ActionAttach, relatedRes.Data(), nil)
+	if policy.IsDeny() {
+		if err == nil {
+			return errorNotFound()
```

Comment:
Resource is available but the request is forbidden due to the policy. Therefore, `Forbidden()` should be better. (However, `BadRequest()` might be safer because error code which is different from above may imply the resource existence.)

Reply:
let's go the secure way. changed to `BadRequest`